src/groovy/org/transmartproject/db/dataquery/highdim/AbstractHighDimensionDataTypeModule.groovy

```diff
@@ -151,6 +153,20 @@ abstract class AbstractHighDimensionDataTypeModule implements HighDimensionDataT
                 "does not support the projection $name")
     }
 
+    @Override
+    List<OntologyTerm> getAllOntologyTermsForDataTypeBy(long resultInstanceId) {
+        I2b2.executeQuery '''
+            select
+                distinct t
+            from QtPatientSetCollection ps, I2b2 t, DeSubjectSampleMapping ssm
+            inner join ssm.platform as p
+            where p.markerType in (:markerTypes)
+                and ssm.conceptCode = t.code
+                and ssm.patient = ps.patient
+                and ps.resultInstance.id = :resultInstanceId
+        ''', [markerTypes : platformMarkerTypes, resultInstanceId: resultInstanceId]
+    }
+
```

Comment:
This should not be a module method. Just fold it into `HighDimensionDataTypeResourceImpl`. The modules handle the data retrieval part that actually changes data type by data type.